pkg/modules/spanpercentile/implspanpercentile/module.go

```diff
@@ -0,0 +1,191 @@
+
+	return transformToSpanPercentileResponse(result)
+}
+
+func transformToSpanPercentileResponse(queryResult *qbtypes.QueryRangeResponse) (*spanpercentiletypes.SpanPercentileResponse, error) {
+	if len(queryResult.Data.Results) == 0 {
+		return nil, errors.New(errors.TypeInvalidInput, errors.CodeInvalidInput, "no data returned from query")
+	}
+
+	scalarData, ok := queryResult.Data.Results[0].(*qbtypes.ScalarData)
+	if !ok {
+		return nil, errors.New(errors.TypeInvalidInput, errors.CodeInvalidInput, "unexpected result type")
+	}
+
+	if len(scalarData.Data) == 0 {
+		return nil, errors.New(errors.TypeInvalidInput, errors.CodeInvalidInput, "no rows returned from query")
+	}
+
+	row := scalarData.Data[0]
+
+	columnMap := make(map[string]int)
+	for i, col := range scalarData.Columns {
+		columnMap[col.Name] = i
+	}
+
+	p50Idx, ok := columnMap["p50_duration_nano"]
+	if !ok {
+		return nil, errors.New(errors.TypeInternal, errors.CodeInternal, "missing p50_duration_nano column")
+	}
+	p90Idx, ok := columnMap["p90_duration_nano"]
+	if !ok {
+		return nil, errors.New(errors.TypeInternal, errors.CodeInternal, "missing p90_duration_nano column")
+	}
+	p99Idx, ok := columnMap["p99_duration_nano"]
+	if !ok {
+		return nil, errors.New(errors.TypeInternal, errors.CodeInternal, "missing p99_duration_nano column")
+	}
+	positionIdx, ok := columnMap["percentile_position"]
+	if !ok {
+		return nil, errors.New(errors.TypeInternal, errors.CodeInternal, "missing percentile_position column")
+	}
+
+	p50, err := toInt64(row[p50Idx])
+	if err != nil {
+		return nil, errors.New(errors.TypeNotFound, errors.CodeNotFound, "no spans found matching the specified criteria")
+	}
+	p90, err := toInt64(row[p90Idx])
+	if err != nil {
+		return nil, errors.New(errors.TypeNotFound, errors.CodeNotFound, "no spans found matching the specified criteria")
+	}
+	p99, err := toInt64(row[p99Idx])
+	if err != nil {
+		return nil, errors.New(errors.TypeNotFound, errors.CodeNotFound, "no spans found matching the specified criteria")
+	}
+	position, err := toFloat64(row[positionIdx])
+	if err != nil {
+		return nil, errors.New(errors.TypeNotFound, errors.CodeNotFound, "no spans found matching the specified criteria")
+	}
+
+	description := fmt.Sprintf("faster than %.1f%% of spans", position)
+	if position < 50 {
+		description = fmt.Sprintf("slower than %.1f%% of spans", 100-position)
+	}
+
+	return &spanpercentiletypes.SpanPercentileResponse{
+		Percentiles: spanpercentiletypes.PercentileStats{
+			P50: p50,
+			P90: p90,
+			P99: p99,
+		},
+		Position: spanpercentiletypes.PercentilePosition{
+			Percentile:  position,
+			Description: description,
+		},
+	}, nil
+}
+
+func toInt64(val any) (int64, error) {
+	switch v := val.(type) {
+	case int64:
+		return v, nil
+	case int:
+		return int64(v), nil
+	case uint64:
+		return int64(v), nil
+	case float64:
+		if v != v {
+			return 0, errors.New(errors.TypeInvalidInput, errors.CodeInvalidInput, "value is NaN")
```

Comment:
even here what will the client do with invalid input, give it a defaul or keep it nil.